build.ps1

```diff
@@ -89,7 +89,7 @@ if(!$PSScriptRoot){
 $TOOLS_DIR = Join-Path $PSScriptRoot "tools"
 $NUGET_EXE = Join-Path $TOOLS_DIR "nuget.exe"
 $CAKE_EXE = Join-Path $TOOLS_DIR "Cake/Cake.exe"
-$NUGET_URL = "https://dist.nuget.org/win-x86-commandline/latest/nuget.exe"
+$NUGET_URL = "https://dist.nuget.org/win-x86-commandline/v4.0.0/nuget.exe"
```

Comment:
I ran into this as well. An alternative which I would recommend is to leave the bootstrapper alone and replace `nuget restore` with `msbuild /t:restore`.
See https://gitter.im/cake-build/cake?at=58d156017b3f37e7541e55b0 and http://blog.nuget.org/20170316/NuGet-now-fully-integrated-into-MSBuild.html